crates/hc_sandbox/tests/cli.rs

```diff
@@ -692,6 +697,150 @@ async fn authorize_zome_call_credentials() {
         .stderr(Stdio::inherit())
         .kill_on_drop(true);
 
+    let mut hc_generate = input_piped_password(&mut cmd).await;
+
+    let launch_info = get_launch_info(&mut hc_generate).await;
+
+    let app_info = get_app_info(
+        launch_info.admin_port,
+        "test-app".into(),
+        *launch_info.app_ports.first().expect("No app ports found"),
+    )
+    .await;
+
+    let mut dna_hashes = match app_info {
+        AppResponse::AppInfo(Some(info)) => {
+            let cell_ids: Vec<Vec<CellInfo>> = info
+                .cell_info
+                .into_iter()
+                .map(|(_, cell_infos)| cell_infos)
+                .collect();
+            println!("cell_ids: {:?}", cell_ids);
+            let dna_hash_1 = match cell_ids[0].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            let dna_hash_2 = match cell_ids[1].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            let dna_hash_3 = match cell_ids[2].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            // The fixture happ contains 3 times the same dna of which 2 have the same dna hash
```

Comment:
Took me a while to figure that one out. Why does the fixture happ contain the exact dna twice? Should that even be allowed for the same happ?

Reply:
That is interesting. I guess that's a matter of answering the question of why that was done in the first place. The second and third roles were added at some point. So which tests uses them and what for? I see one test is using properties and a different network seed but I don't know what the point of the same DNA twice with the same hash would be and I expect that even though Holochain is accepting the install, it's probably handling that wrong internally.

Reply:
Yeah, guess who added that role. Myself of course for the roles settings test. I'm not sure it's actually needed at all. But it *does* seem like a loophole that's not covered currently if it allows installing the exact same cell twice for a happ.

Reply:
Agreed: https://github.com/holochain/holochain/issues/5064